src/index.ts

```diff
@@ -142,6 +142,7 @@ export enum SimpleRuleType {
 export interface EngineRuleData<V, T> {
   type: T;
   value: V;
+  caseSensitive?: boolean;
```

Comment:
I'm not sure if we need this extra parameter since it add complexity everywhere.

Reply:
Pour lui, il n'y a pas de soucis c'est juste une ligne du cote EM pour pouvoir l'utiliser dans le créateur des queries avec regex. Il n'a pas d'incidence. Juste pour le type checking si il y'a la presence de cette donnée en trop.